src/tile.cpp

```diff
@@ -45,53 +45,57 @@ void Tile::addFeature(Tile& tile,
-            return;
+        if (feature.geometry.is<ProjectedPoints>()) {
+            auto const& points = feature.geometry.get<ProjectedPoints>();
+            if (!points.empty()) {
+                auto pt_size = points.size();
+                tile.numPoints += pt_size;
+                tile.numSimplified += pt_size;
+                tile.features.emplace_back(points, type, feature.tags);
+            }
         }
-
     } else {
-        simplified.set<ProjectedRings>();
-
-        // simplify and transform projected coordinates for tile geometry
-        for (auto const& ring : feature.geometry.get<ProjectedRings>()) {
-            // filter out tiny polylines & polygons
-            if (!noSimplify &&
-                ((type == ProjectedFeatureType::LineString && ring.dist < tolerance) ||
-                 (type == ProjectedFeatureType::Polygon && ring.area < sqTolerance))) {
-
-                tile.numPoints += ring.points.size();
-                continue;
-            }
-
-            ProjectedRing simplifiedRing;
+        if (feature.geometry.is<ProjectedRings>()) {
+            auto const& rings = feature.geometry.get<ProjectedRings>();
+            if (noSimplify) {
+                if (!rings.empty()) {
+                    auto rings_size = rings.size();
+                    tile.numPoints += rings_size;
+                    tile.numSimplified += rings_size;
+                    tile.features.emplace_back(rings, type, feature.tags);
+                }
+            } else {
+                ProjectedRings p_rings;
+                // simplify and transform projected coordinates for tile geometry
+                for (auto const& ring : rings) {
+                    // filter out tiny polylines & polygons
+                    if ((type == ProjectedFeatureType::LineString && ring.dist < tolerance) ||
+                         (type == ProjectedFeatureType::Polygon && ring.area < sqTolerance)) {
+                        tile.numPoints += ring.points.size();
+                        continue;
+                    }
+
+                    ProjectedRing simplifiedRing;
+                    for (auto const& p : ring.points) {
+                        // keep points with importance > tolerance
+                        if (p.z > sqTolerance) {
+                            simplifiedRing.points.emplace_back(p);
+                            ++tile.numSimplified;
+                        }
+                        ++tile.numPoints;
+                    }
+                    p_rings.emplace_back(std::move(simplifiedRing));
+                }
 
-            for (auto const& p : ring.points) {
-                // keep points with importance > tolerance
-                if (noSimplify || p.z > sqTolerance) {
-                    simplifiedRing.points.push_back(p);
-                    ++tile.numSimplified;
+                if (!p_rings.empty()) {
+                    tile.features.emplace_back(std::move(p_rings), type, feature.tags);
```

Comment:
@mourner `std::move` should be avoided when it will prevent RVO. In this case I think `std::move` is needed to trigger a move rather than a copy. However even if this is not needed for a given compiler I don't seen any harm, but would be happy to be wrong if @artemp sees this differently.